File: 012/Problem12.py
import time
import math


def countDivisors(n):
    count = 0
    for i in range(1, int(math.sqrt(n))):
        if n % i == 0:
            if n / i == i:
                count += 1
            else:
                count = count + 2
    return count


def problem():
    numDivisors = 500
    i = 3
    while True:
        key = i
        su = 0
        while i > 0:
            su += i
            i -= 1
        if countDivisors(su) > numDivisors:
            print(su)
            break
        i = key + 1


start = time.time()
problem()
end = time.time()
print("Function took " + str(end - start))

File: 012/Problem12.cs
using System;

namespace EulerProblems
{
    class Program
    {
        static void Main(string[] args)
        {
            var watch = new System.Diagnostics.Stopwatch();

            watch.Start();
            Problem();
            watch.Stop();

            Console.WriteLine($"Function Took: {watch.Elapsed}");
        }

        private static int CountDivisors(int n)
        {
            int count = 0;
            for (int i = 1; i <= Math.Sqrt(n); i++)
            {
                if (n % i == 0)
                {
                    if (n / i == i)
                        count++;

                    else
                        count = count + 2;
                }
            }
            return count;
        }
        private static void Problem()
        {
            var NumDivisors = 500;
            var i = 3;
            while (true)
            {
                var key = i;
                var sum = 0;
                while (i > 0)
                {
                    sum += i;
                    i--;
                }
                if (CountDivisors(sum) > NumDivisors)
                {
                    Console.WriteLine(sum);
                    break;
                }

                i = key + 1;
            }
        }
    }
}